### PR TITLE
Cleaning up pool position logic (fixing errors)

### PR DESCRIPTION
**Pull request by IanWoodard**
TLDR: I cleaned up the logic in the pool position widget and hopefully reduced the number of API calls being sent out to etherscan.

I noticed some errors in the console due to rate limiting from etherscan resulting in a string as result rather than the array we expected. To fix this, I added a guard clause to ensure the result is an array before we try to do anything with it. With that being said, this does not address the underlying problem which is that the etherscan API is being called enough to rate limit us. After doing a good bit of looking into it, I wasn't able to come up with a perfect solution, but I did clean up the logic a good deal and got it to send fewer API calls for me locally (although it may be different for different people). One thing I noticed is that when the screen is resized passed certain breakpoints (used for responsiveness), the component is re-rendered (as it should be), but more API calls are also sent out (which is not ideal). I do not think this poses a big issue but would like to continue to think about ways to prevent this (without just adding even more checks).

### CODE REVIEW

File: src/pages/BlendPoolPage.tsx [hunk 1]
```diff
@@ -39,6 +39,15 @@ type PoolParams = {
...
 
+export type AccountData = {
+  address: string;
+  connector: Connector<any, any> | undefined;
+    ens: {
+        avatar: string | null | undefined;
+        name: string;
+    } | undefined;
+}
+
```
Comment (adamegyed) [inline src/pages/BlendPoolPage.tsx:50]: Is this exported by `wagmi` anywhere? Ideally we shouldn't need to redefine it here.

Comment (adamegyed) [reply src/pages/BlendPoolPage.tsx:50]: Actually I looked through the `wagmi` 0.4.x docs and it looks like the type declaration is different in that the ENS is not explicitly included in the account data. Since we're not using the ENS anywhere from this type declaration, could we drop this declaration and use the `wagmi` one instead?

**Review by adamegyed — DISMISSED** [review]
Good catch on this. Tested locally and this successfully deduplicated etherscan requests. As for the other duplicate requests, I believe most originate from the `BlendPoolContext`'s data population method. I will investigate further, and the rerendering problem should also be addressable by a React context.

Just have one requested change for a type declaration duplicate.

File: src/pages/BlendPoolPage.tsx (hunk 1 above)
Comment (IanWoodard) [reply src/pages/BlendPoolPage.tsx:50]: Could you point me to the type definition in the docs? I can't seem to find it

Comment (haydenshively) [reply src/pages/BlendPoolPage.tsx:50]: Neither `wagmi` 0.4.x nor 0.5.x export the `GetAccountResult` type that is returned by the `getAccount` action (the one used by `useAccount` hook). To be more precise, they are exported from `index.ts` but don't make their way to the npm package. Unclear if this is due to misconfigured `package.json` or the fact that `GetAccountResult` is dependent on a generic type constraint that _doesn't_ get exported... but regardless Ian's solution is the best one here

**Review by haydenshively — APPROVED** [review]
